app/src/main/java/org/literacyapp/authentication/StudentImageCollectionActivity.java

```diff
@@ -131,71 +149,77 @@ public void onCameraViewStopped() {
-                    Rect[] faces = ppF.getFacesForRecognition();
-                    if ((faces != null) && (faces.length == 1)) {
-                        faces = MatOperation.rotateFaces(imgRgba, faces, ppF.getAngleForRecognition());
-                        face = faces[0];
-                        faceDetected = true;
-                        // Reset startTimeFallback for fallback timeout, because at least one face has been detected
-                        startTimeFallback = currentTime;
-                        isFaceInsideFrame = DetectionHelper.isFaceInsideFrame(animalOverlay, imgRgba, face);
-
-                        if (isFaceInsideFrame){
-                            studentImages.add(img);
-
-                            if(DIAGNOSE_MODE) {
-                                MatOperation.drawRectangleAndLabelOnPreview(imgRgba, face, "Face detected", true);
-                            }
-
-                            // Stop after NUMBER_OF_IMAGES (settings option)
-                            if(imagesProcessed == NUMBER_OF_IMAGES){
-                                storeStudentImages();
-                                finish();
+        if (authenticationAnimationAlreadyPlayed || ((startTimeAuthenticationAnimation + AUTHENTICATION_ANIMATION_TIME) < currentTime)){
+            prepareForAuthentication();
+
+            Mat imgCopy = new Mat();
+
+            // Store original image for face recognition
+            imgRgba.copyTo(imgCopy);
+
+            // Mirror front camera image
+            Core.flip(imgRgba,imgRgba,1);
+
+            Rect face = new Rect();
+            boolean isFaceInsideFrame = false;
+            boolean faceDetected = false;
+
+            if((lastTime + TIMER_DIFF) < currentTime){
+                lastTime = currentTime;
+                List<Mat> images = ppF.getCroppedImage(imgCopy);
+                if(images != null && images.size() == 1){
+                    Mat img = images.get(0);
+                    if(img != null) {
+                        Rect[] faces = ppF.getFacesForRecognition();
+                        if ((faces != null) && (faces.length == 1)) {
+                            faces = MatOperation.rotateFaces(imgRgba, faces, ppF.getAngleForRecognition());
+                            face = faces[0];
+                            faceDetected = true;
+                            // Reset startTimeFallback for fallback timeout, because at least one face has been detected
+                            startTimeFallback = currentTime;
+                            isFaceInsideFrame = DetectionHelper.isFaceInsideFrame(animalOverlay, imgRgba, face);
+
+                            if (isFaceInsideFrame){
+                                mediaPlayerAnimalSound.start();
+                                studentImages.add(img);
+
+                                if(DIAGNOSE_MODE) {
+                                    MatOperation.drawRectangleAndLabelOnPreview(imgRgba, face, "Face detected", true);
```

Comment:
Since this will not be used in production, we don't need to translate "Face detected" to Swahili, right?